src/BSON/Int64.c

```diff
@@ -265,11 +265,23 @@ static zend_object* php_phongo_int64_clone_object(phongo_compat_object_handler_t
 	return new_object;
 }
 
-static int php_phongo_int64_compare_objects(zval* o1, zval* o2)
+static bool php_phongo_int64_is_int64_object(zval* object)
 {
-	php_phongo_int64_t *intern1, *intern2;
+	if (Z_TYPE_P(object) != IS_OBJECT) {
+		return false;
+	}
 
-	ZEND_COMPARE_OBJECTS_FALLBACK(o1, o2);
+	return Z_OBJ_P(object)->ce == php_phongo_int64_ce;
+}
+
+static bool php_phongo_int64_is_long_or_double(zval* value)
```

Comment:
How is this function called?

Reply:
This function is used in an `assert` call in `php_phongo_int64_compare_with_long_or_float` a few lines below.